pkg/endpoints/cluster.go

```diff
@@ -162,3 +164,70 @@ func (r Resource) GetProperties(request *restful.Request, response *restful.Resp
 	properties := Properties{InstallNamespace: os.Getenv("INSTALLED_NAMESPACE")}
 	response.WriteEntity(properties)
 }
+
+// Get dashboard version
+func (r Resource) GetDashboardVersion(request *restful.Request, response *restful.Response) {
+	config, err := rest.InClusterConfig()
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	properties := Properties{InstallNamespace: os.Getenv("INSTALLED_NAMESPACE")}
+	version := ""
+
+	api1 := clientset.AppsV1()
+	listOptions := metav1.ListOptions{
+		LabelSelector: "app=tekton-dashboard",
+	}
+	deployments, err := api1.Deployments(properties.InstallNamespace).List(listOptions)
+
+	for _, deployment := range deployments.Items {
+		deploymentLabels := deployment.GetLabels()
+		version = deploymentLabels["version"]
+	}
+
+	if version == "" {
+		response.WriteEntity("Unknown")
+		return
+	}
+	response.WriteEntity(version)
+	return
+}
+
+// Get pipelines version
+func (r Resource) GetPipelineVersion(request *restful.Request, response *restful.Response) {
```

Comment:
Same comments for the previous one, we could potentially combine this with the previous (and change the selector?) - we'd reduce plenty code dupe

Reply:
Have returned an error if there is an error in getting the deployment
Have changed the errors, same as above

It's not just the selectors which are different, it's the namespace which it searches as well as when the PR for openshift pipelines comes in it'll also be checking for that install as well